### PR TITLE
UCP/RNDV/MLANE: disable mlane on self/shared AM TL

### PR DESCRIPTION
**Pull request by hoopoepg**

### CODE REVIEW

**Comment by shamisp** [comment]
How did you manage to get multiple lanes with self ?

**Comment by hoopoepg** [comment]
in case if self or shm transport used to communication, then single lane is used inside of UCP EP for AM or RMA proto

**Comment by hoopoepg** [comment]
failed due to FS issue

**Comment by hoopoepg** [comment]
bot:bgate:retest

**Comment by shamisp** [comment]
But was it opening more than one lane ? 
I think it is not supposed to since there is only one "shm" device.

**Comment by hoopoepg** [comment]
mlx test failed due to git issue

**Comment by hoopoepg** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7162/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7163/ for details (Mellanox internal link).

**Comment by shamisp** [comment]
ok

File: src/ucp/wireup/select.c [hunk 1]
```diff
@@ -1031,8 +1037,13 @@ static ucs_status_t ucp_wireup_add_am_bw_lanes(ucp_ep_h ep, const ucp_ep_params_
             bw_info.md_map            |= UCS_BIT(context->tl_rscs[rsc_index].md_index);
             bw_info.local_dev_bitmap  &= ~UCS_BIT(context->tl_rscs[rsc_index].dev_index);
             bw_info.remote_dev_bitmap &= ~UCS_BIT(address_list[addr_index].dev_index);
-            break; /* do not continue searching due to we found
-                      AM lane (and there is only one lane) */
+            if (ucp_wireup_is_ep_single_lane(ep, rsc_index)) {
+                /* if AM lane is SELF or SHMEM - then do not use more lanes */
+                return UCS_OK;
```
Comment (yosefe) [inline src/ucp/wireup/select.c]: so it mean before the fix we could have multi-lane with SHM as well?
also, do we need similar fix for rma lanes as well?

Comment (hoopoepg) [reply src/ucp/wireup/select.c]: i think yes, i found this issue investigating hang on OMPI finalize

pick it up to 1.4?

Comment (yosefe) [reply src/ucp/wireup/select.c]: yes. should we fix also rma lanes?

Comment (hoopoepg) [reply src/ucp/wireup/select.c]: RMA is fixed in this PR as well

**Pull request merged**